viewing_party/party.py

```diff
@@ -1,23 +1,167 @@
+    # loop through most watched dict to update
+    for genre, count in most_watched_genre_dict.items():
+        if count > highest_count_sofar:
+            highest_count_sofar = count
+            highest_genre = genre
+    #this better work
+    return highest_genre           
+            
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def get_unique_watched(user_data):
+
+    user_unique_movies = []  # a list of dictionaries
+    movies_titles_friends_watched = []  # a list of strings
+
+    for item in user_data.get("friends"):
+        for x in item.get("watched"):
+            movies_titles_friends_watched.append(x.get("title"))
+    
+    for item in user_data.get("watched"):
+        movie_user_has_watched = item
+        if movie_user_has_watched.get("title") not in movies_titles_friends_watched:
+                user_unique_movies.append(movie_user_has_watched)
+
+    return user_unique_movies
+
+def get_friends_unique_watched(user_data):
+
+    friends_movies = [] # a list of dictionaries
+    movies_titles_user_watched = [] # a list of strings
+
+    for item in user_data.get("watched"):
+        movies_titles_user_watched.append(item.get("title"))
+
+    for item in user_data.get("friends"):
+        for x in item.get("watched"):
+            if x.get("title") not in movies_titles_user_watched:
+                friends_movies.append(x)
+
+    new_list = []
+    for dictionary in friends_movies:
+        if dictionary not in new_list:
+            new_list.append(dictionary)
+    
+    return new_list
+
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+
+    friends_unique_watched = get_friends_unique_watched(user_data)
+    print(friends_unique_watched)
+
+    movie_recommendations = []
+
+    for x in friends_unique_watched:
+        if x.get("host") in user_data.get("subscriptions"):
+            movie_recommendations.append(x)
+
+    return movie_recommendations
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(user_data):
+
+    # Consider the user's most frequently watched genre.
+
+    frequently_watched_genre = []
+    for item in user_data.get("watched"):
+        frequently_watched_genre.append(item.get("genre"))
+    
+    # Here I converted from list to set, then again to list to get rid of the duplicates
+    frequently_watched_genre_set = set(frequently_watched_genre)
+    frequently_watched_genre_list = list(frequently_watched_genre_set)
+
+        
+    friends_recommendations_list_of_dict = get_friends_unique_watched(user_data)
+
+
+    movie_recommendations_genre = []
+    for i in friends_recommendations_list_of_dict:
+        if i.get("genre") == frequently_watched_genre_list:

```

Comment:
Let's come up with a variable name that's more descriptive than `i`

Reply:
What _is_ `frequently_watched_genre_list`? Is it the most watched genre type, or is it a list? Can we compare a string to a list? Or, should `frequently_watched_genre_list` be something else?

What if we call one of our functions to get us the most watched genre type, then use it to compare against our recommendation list?